sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java

```diff
@@ -434,14 +434,21 @@ public Mono<String> captureAsync(String containerName, String vhdPrefix, boolean
             .serviceClient()
             .getVirtualMachines()
             .captureAsync(this.resourceGroupName(), this.name(), parameters)
-            .map(
-                captureResultInner -> {
-                    try {
-                        return SerializerUtils.getObjectMapper().writeValueAsString(captureResultInner);
-                    } catch (JsonProcessingException ex) {
-                        throw logger.logExceptionAsError(Exceptions.propagate(ex));
-                    }
-                });
+            .map(this::serializeCaptureResult);
+    }
+
+    private String serializeCaptureResult(VirtualMachineCaptureResultInner captureResultInner) {
+        Map<String, Object> resultMap = new HashMap<>();
+        resultMap.put("id", captureResultInner.id());
+        resultMap.put("contentVersion", captureResultInner.contentVersion());
+        resultMap.put("schema", captureResultInner.schema());
+        resultMap.put("resources", captureResultInner.resources());
```

Comment:
`parameters` property?

Is it possible to add a test to verify all property is serialized.

I kind of forget why we do this serialization..

Reply:
Right, I didn't serialize this property.. 

> Is it possible to add a test to verify all property is serialized.

It's possible, though I wonder if it's necessary. 
The string is like a convenience layer wrapper of the inner model. We don't have to ensure all newly added inner model's property is exposed in the wrapper(and we don't even document what properties are in there..).
If some customer really needs the properties be up-to-date, we can add for them afterwards.

> I kind of forget why we do this serialization..

Yeah, I'm surprised that we return string instead of the model itself..

Reply:
My guess why we serialize these read-only properties is that some customer wants the serialized information of the captured result, and the model itself doesn't have this ability(since all its properties are marked JSON write-only).